### PR TITLE
Feature/4 create mongoose user schema

### PR DESCRIPTION
**Pull request by davideastmond**
First draft of UserSchema as per Figma. To be merged into development branch of backend-repo
Resolves this [ticket](https://github.com/chingu-voyages/v25-bears-team-05/issues/4)

### CODE REVIEW

File: api/src/models/user/user.schema.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  avatar: {
+    type: [{url: String}],
+  },
+  connections: {},
+  connectionOf: {},
+  threads: {
+    started:{},
+    commented: {},
+    liked: {},
+    shared:{}
```
Comment (davideastmond) [inline api/src/models/user/user.schema.ts:29]: I think there is a better way to type these if we decide to implement this feature.

File: api/src/models/user/user.types.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    oauth?: string,
+  };
+  avatar: [ { url: string } ];
+  connections: any;
+  connectionOf: any;
+  threads: {
+    started: {},
+    commented: {},
+    liked: {},
+    shared: {},
```
Comment (davideastmond) [inline api/src/models/user/user.types.ts:18]: see above

File: api/src/models/user/user.schema.ts (hunk 1 above)
Comment (lily-law) [reply api/src/models/user/user.schema.ts:29]: Let come back to look at this after the prototype. We could look over our user actions flow and work out what would be best.

**Review by lily-law — APPROVED** [review]
Everything looks good to me on this, nice and clean, and I'm glad we're starting out with unit tests, nice work! 

Just a note: removing the api/src/models/User.ts has broken the imports in api/src/routes/api/

**Comment by davideastmond** [comment]
> Everything looks good to me on this, nice and clean, and I'm glad we're starting out with unit tests, nice work!
> 
> Just a note: removing the api/src/models/User.ts has broken the imports in api/src/routes/api/

Nice catch! A lot of stuff was broken because of the renaming of the user model files. All looks good now.

**Pull request merged**